rs/ic-observability/multiservice-discovery/tests/tests.rs

```diff
@@ -0,0 +1,144 @@
+// Note: If you need to update the assertions, please check the README.md file for guidelines.
+#[cfg(test)]
+mod tests {
+    use assert_cmd::cargo::CommandCargoExt;
+    use multiservice_discovery_shared::builders::prometheus_config_structure::{
+        PrometheusStaticConfig, IC_NAME, IC_SUBNET, JOB,
+    };
+    use tempfile::tempdir;
+    use std::collections::{BTreeMap, BTreeSet};
+    use std::process::Command;
+    use std::time::Duration;
+    use std::thread;
+    use tokio::runtime::Runtime;
+
+    const BAZEL_SD_BIN: &str = "rs/ic-observability/multiservice-discovery/multiservice-discovery";
+    const CARGO_SD_BIN: &str = "multiservice-discovery";
+
+    async fn fetch_targets() -> anyhow::Result<Vec<PrometheusStaticConfig>> {
+        let timeout_duration = Duration::from_secs(500);
+        let start_time = std::time::Instant::now();
+
+        loop {
+            if start_time.elapsed() > timeout_duration {
+                return Err(anyhow::anyhow!("Timeout reached"));
+            }
+            if let Ok(response) = reqwest::get("http://localhost:8000/prom/targets").await {
+                if response.status().is_success() {
+                    let text = response.text().await?;
+                    let targets: Vec<PrometheusStaticConfig> = serde_json::from_str(&text).unwrap();
+                    return Ok(targets);
+                }
+            }
+            thread::sleep(Duration::from_secs(5));
+        }
+    }
+
+    #[test]
+    fn prom_targets_tests() {
+        let rt = Runtime::new().unwrap();
+        let registry_dir = tempdir().unwrap();
+        registry_dir.path().to_str().unwrap();
+        let args = vec![
+            "--targets-dir",
+            registry_dir.path().to_str().unwrap(),
+        ];
+
+        let mut sd_server = Command::cargo_bin(CARGO_SD_BIN)
+            .unwrap_or(Command::new(BAZEL_SD_BIN))
+            .args(args)
+            .spawn()
+            .unwrap();
+
+        let targets = rt.block_on(rt.spawn(async { 
+            fetch_targets().await 
+        })).unwrap().unwrap();
+        sd_server.kill().unwrap();
+
+        let labels_set =
+            targets
+                .iter()
+                .cloned()
+                .fold(BTreeMap::new(), |mut acc: BTreeMap<String, BTreeSet<String>>, v| {
+                    for (key, value) in v.labels {
+                        if let Some(grouped_set) = acc.get_mut(&key) {
+                            grouped_set.insert(value);
+                        } else {
+                            let mut new_set = BTreeSet::new();
+                            new_set.insert(value);
+                            acc.insert(key, new_set);
+                        }
+                    }
+                    acc
+                });
+
+        assert_eq!(targets.len(), 7374);
```

Comment:
This will be painful to update all the time, and will lead to unpredictably and uncontrollably failing tests. We'll have to find a better way to do this. For instance, add an error margin, and periodically and automatically update the expected value, based on the reference value provided by the code from "main". But this can come in follow-up PRs, I believe.